lib/log.c

```diff
@@ -155,7 +156,11 @@ unsigned int log_get_level() {
 	return loglevel;
 }
 
-int log_set_logfile(const char *logfile) {
+int log_reopen_file() {
+	if (fp_logfile) {
+		fclose(fp_logfile);
```

Comment:
I'd prefer to use `freopen()` instead of closing and reopening the log file.
That way, there's no need to hold the log file name in a global variable.

Reply:
`freopen` too takes a filename. And the file needs to be reopened by name, as after a logrotate there's an entirely new file under the name. 

`freopen` with a pathname set to NULL merely changes the mode of the already open file, but will still refer to the same file, which will by then have moved to `daliserver.log.1`.

Reply:
It's unfortunate that the [POSIX specification](https://pubs.opengroup.org/onlinepubs/9699919799/functions/freopen.html) is so vague in terms of how `freopen(NULL,...)` should be handled.

glibc actually _does_ reopen and close the file. However, it does so by opening `/proc/self/fd/<original file descriptor>`, which evidently causes the Linux kernel to merely duplicate the file kernel handle plus the changed attributes. If the directory entry has changed in the meantime, this will not be reflected by the new handle. Fun fact: The kernel creates a new file descriptor, but freopen calls dup2() to reuse the old one.

Apparently, some other UNIXes don't even support this at all.